packages/@aws-cdk/cicd/lib/pipeline-deploy-stack-action.ts

```diff
@@ -0,0 +1,84 @@
+
+import cfn = require('@aws-cdk/aws-cloudformation');
+import api = require('@aws-cdk/aws-codepipeline-api');
+import cdk = require('@aws-cdk/cdk');
+import cxapi = require('@aws-cdk/cx-api');
+
+export interface PipelineDeployStackActionProps {
+  /**
+   * The CDK stack to be deployed.
+   */
+  stack: cdk.Stack;
+  /**
+   * The CodePipeline stage in which to perform the deployment.
+   */
+  stage: api.IStage;
+  /**
+   * The CodePipeline artifact that holds the synthesized app.
```

Comment:
Provide some more details here on how to produce this ("the output of `cdk synth -o`")